src/dupin.cpp

```diff
@@ -0,0 +1,176 @@
+#include "dupin.h"
+#include <Eigen/Dense>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+#include <tbb/blocked_range2d.h>
+#include <tbb/global_control.h>
+#include <tbb/parallel_for.h>
+#include <unordered_map>
+#include <vector>
+
+using namespace std;
+using namespace Eigen;
+
+DynamicProgramming::DynamicProgramming()
+    : num_bkps(0), num_parameters(0), num_timesteps(0), jump(1), min_size(3) {}
```

Comment:
```suggestion
    : num_bkps(1), num_parameters(0), num_timesteps(0), jump(1), min_size(3) {}
```
We also should set size by the signal passed, not explicit setting.

Reply:
Yeah this is just the default constructor; honestly probably won't be used